### ISSUE TITLE
Observing PromiseArrays doesn't seem to work

### ISSUE THREAD

**Issue by courthead** [emberjs/data#1557]
TL;DR: An observer like `books.@each.libraries.isSettled` won't fire when `isSettled` changes.

http://jsbin.com/iLomeVU/4/edit

In the JS Bin there are two models: books and libraries. Each model has an async `hasMany` relationship to the other. There's also an `ObjectController` whose content is a library, and a corresponding template that shows a list of the books in the library, which have to be loaded from the server.`

Immediately after the server responds with the list of books, each book's `libraries` attribute is a `PromiseArray` where`isFulfilled` is `false`. Once each promise settles, `isFulfilled` should change to true.

This change should trigger an update to a property that observes `books.@each.libraries.isFulfilled`. Unfortunately, it doesn't. You can see in the output that even though each project's libraries are fulfilled, the property is never updated.
